pkg/apis/eventing/v1alpha1/broker_validation.go

```diff
@@ -20,23 +20,30 @@ import (
 	"context"
 	"fmt"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	"knative.dev/eventing-rabbitmq/pkg/utils"
 	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
 	"knative.dev/pkg/apis"
-	"knative.dev/pkg/apis/duck"
 	"knative.dev/pkg/kmp"
+	"knative.dev/pkg/webhook/resourcesemantics"
 )
 
-// +k8s:deepcopy-gen=false
+// stub Broker in order to set up validations and defaults
+// +k8s:controller-gen=false
 type RabbitBroker struct {
```

Comment:
So now if the eventing.Broker type changes we have to update this ourselves?

Reply:
Yeah both the RabbitBroker and RabbitTrigger will have to track the core API. Spec/Status changes are fine but when the version gets bumped, it'll need changing here and elsewhere in the codebase